src/base/http/connection.h

```diff
@@ -27,41 +28,69 @@
  * exception statement from your version.
  */
 
-
 #pragma once
 
 #include <QElapsedTimer>
 #include <QObject>
 
+#include "types.h"
+
 class QTcpSocket;
 
 namespace Http
 {
-    class IRequestHandler;
-    struct Response;
-
     class Connection : public QObject
     {
         Q_OBJECT
         Q_DISABLE_COPY(Connection)
 
     public:
-        Connection(QTcpSocket *socket, IRequestHandler *requestHandler, QObject *parent = nullptr);
-        ~Connection();
+        enum Status
+        {
+            PendingRequest,
+            PendingStatus,
+            PendingHeaders,
+            PendingContent,
+            WritingContent,
+            Closed
+        };
+
+        // Connection takes ownership of socket
+        explicit Connection(QTcpSocket *socket, QObject *parent = nullptr);
 
-        bool hasExpired(qint64 timeout) const;
-        bool isClosed() const;
+        void sendStatus(const ResponseStatus &status);
+        void sendHeaders(const HeaderMap &headers);
+        void sendContent(const QByteArray &data);
+        void close();
+
+        Http::Environment enviroment() const;
+        Status status() const;
+        qint64 inactivityTime() const;
+        qint64 bytesToWrite() const;
+
+    signals:
+        void bytesWritten(qint64 bytes);
+        void disconnected();
+        void readyRequest(Request request);
 
     private slots:
         void read();
 
     private:
-        static bool acceptsGzipEncoding(QString codings);
-        void sendResponse(const Response &response) const;
+        enum ConnectionType
+        {
+            Unknown,
+            KeepAlive,
+            ClosedConnection
```

Comment:
`Closed` does not look like the correct connection type. Maybe `Temporary`, or something else?

Reply:
Guys, read the references (rfc or MDN or wiki) when dealing with networking terms.
https://en.wikipedia.org/wiki/HTTP_persistent_connection
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Connection
Here it can be `close`.

Reply:
>Here it can be `close`

Are you sure you really talk about connection **type** and not the **state**? So connection can be established as "closed connection"?

Reply:
Or do you want to directly use `Connection` header value as the connection type? IMO, this is incorrect (it produces an inappropriate/inconsistent name).
In addition, it is highly controversial to use the term "connection type" here. IIRC, client or server can send `Connection: close` at any time (in any request/response), even if previous requests/responses within the same connection had `Connection: keep-alive`. Of course, you can say that the connection just changed its type, but "change the type" is nonsense in programming terminology, IMO.

Reply:
I would avoid the term "connection type" altogether, and instead have just a flag indicating that the connection should be closed after the response, set to `false` by default (because the Connection is keep-alive by default in HTTP 1.1), and set to `true` as soon as the Request with `Connection: close` is received.

Reply:
It looks like I'm late with my last comment and you've already changed it in a similar way.